gh-99304: [Enum] clarify what constitutes a flag alias (#99395)
python/cpython: Doc/library/enum.rst

   in most of the same places that a string can be used.  The result of any string
   operation performed on or with a *StrEnum* member is not part of the enumeration.

   .. note::

      There are places in the stdlib that check for an exact :class:`str`
      instead of a :class:`str` subclass (i.e. ``type(unknown) == str``
      instead of ``isinstance(unknown, str)``), and in those locations you
      will need to use ``str(StrEnum.member)``.

   .. note::

      Using :class:`auto` with :class:`StrEnum` results in the lower-cased member
      name as the value.

   .. note::

>       :meth:`__str__` is :func:`str.__str__` to better support the

Reviewer: ```suggestion
      :meth:`~object.__str__` is :meth:`!str.__str__` to better support the
```

* Fix broken link to abstract `__str__`
* Don't attempt to resolve `str.__str__` to avoid a failed resolution and a warning
* Use appropriate role on `str.__str__`

>       *replacement of existing constants* use-case.  :meth:`__format__` is likewise
>       :func:`str.__format__` for that same reason.

Reviewer: ```suggestion
      *replacement of existing constants* use-case.  :meth:`~object.__format__` is likewise
      :meth:`!str.__format__` for that same reason.
```

Fix it here too

   .. method:: __iter__(self):

      Returns all contained non-alias members::

         >>> list(Color.RED)
         [<Color.RED: 1>]
         >>> list(purple)
         [<Color.RED: 1>, <Color.BLUE: 4>]

>       ..versionchanged:: 3.11  Aliases are no longer returned during iteration.

Reviewer: ```suggestion
      .. versionchanged:: 3.11
      
         Aliases are no longer returned during iteration.
```

Fix directive syntax

         >>> ~Color.RED
         <Color.GREEN|BLUE: 6>

>       ..versionchanged:: 3.11 inversion no longer returns a flag with a negative
>       value

Reviewer: ```suggestion
      .. versionchanged:: 3.11
      
      Inversion of a flag now returns a positive value that is the
      union of all flags not in the given flag, rather than a negative value.
```

Fix the syntax, and be clearer and more explicit here

      Using :class:`auto` with :class:`IntFlag` results in integers that are powers
      of two, starting with ``1``.

   .. versionchanged:: 3.11

>       :meth:`__str__` is now :func:`int.__str__` to better support the

Reviewer: ```suggestion
      :meth:`~object.__str__` is now :meth:`!int.__str__` to better support the
```

Fix `__str__` link, `int.__str__` role and de-resolve as above.

>       *replacement of existing constants* use-case.  :meth:`__format__` was
>       already :func:`int.__format__` for that same reason.

Reviewer: ```suggestion
      *replacement of existing constants* use-case.  :meth:`~object.__format__` was
      already :meth:`!int.__format__` for that same reason.
```

Likewise

>       Inversion of a *IntFlag* now returns a positive value that is the

Reviewer: ```suggestion
      Inversion of a :class:`!IntFlag` now returns a positive value that is the
```

`IntFlag` is a class rather than a parameter (per the [docs style guide](https://devguide.python.org/documentation/markup/#quick-reference)).

Author: Thanks for the link, I'll give it a read.  Looks like some serious redoing of the enum docs in another PR.

      union of all flags not in the given flag, rather than a negative value.

>       This matches the existing *Flag* behavior.

Reviewer: ```suggestion
      This matches the existing :class:`Flag` behavior.
```

Likewise
